docs/legacy/legacy-upgrade-guide.md

```diff
@@ -106,7 +106,7 @@ The `Texture2DArray` class has been removed as its utility was limited and the s
 
 #### FenceSync Class Removed
 
-The `FenceSync` class has been removed as its utility was limited. It may be added back in the future if a clear use case arises. If required, syncing can be done directly through the `WebGLFenceSync` object.
+The old `FenceSync` class has been removed. Applications can use the new portable [`Fence`](/docs/api-reference/core/resources/fence) resource created via `device.createFence()` to wait for GPU work to complete.
```

Comment:
Nice to restore another feature post-v9. Sounds like it wasn't used before very often, but it might be helpful for picking?